src/Dialog.cc

```diff
@@ -75,3 +78,153 @@ QQuickItem *Dialog::RootItem() const
   return dialogItem;
 }
 
+/////////////////////////////////////////////////
+template <typename T>
+bool Dialog::UpdateConfigAttribute(const std::string &_path,
+  const std::string &_attribute, const T &_value) const
+{
+  if (_path.empty())
+  {
+    ignerr << "Missing config file" << std::endl;
+    return false;
+  }
+
+  std::string configFull = _path;
+
+  // Use tinyxml to read config
+  tinyxml2::XMLDocument doc;
+  auto success = !doc.LoadFile(configFull.c_str());
+  if (!success)
+  {
+      ignerr << "Failed to load file [" << configFull << "]: XMLError"
+             << std::endl;
+    return false;
+  }
+
+  // Update attribute value for the correct dialog
+    for (auto dialogElem = doc.FirstChildElement("dialog");
+      dialogElem != nullptr;
+      dialogElem = dialogElem->NextSiblingElement("dialog"))
+  {
+    if(dialogElem->Attribute("name") == this->objectName().toStdString())
+    {
+      dialogElem->SetAttribute(_attribute.c_str(), _value);
+    }
+  }
+
+  // Write config file
+  tinyxml2::XMLPrinter printer;
+  doc.Print(&printer);
+
+  std::string config = printer.CStr();
+  std::ofstream out(_path.c_str(), std::ios::out);
+  if (!out)
+  {
+    std::string str = "Unable to open file: " + _path;
+    str += ".\nCheck file permissions.";
+  }
+  else
+    out << config;
+
+  return true;
+}
+
+/////////////////////////////////////////////////
+void Dialog::SetDefaultConfig(const std::string &_config)
+{
+  this->dataPtr->config = _config;
+}
+
+/////////////////////////////////////////////////
+std::string Dialog::ReadAttributeValue(const std::string &_path,
+  const std::string &_attribute) const
+{
+  tinyxml2::XMLDocument doc;
+  std::string value {""};
+  std::string config = "<?xml version=\"1.0\"?>\n\n";
+  tinyxml2::XMLPrinter defaultPrinter;
+  bool configExists{true};
+
+  // Check if the passed in config file exists.
+  // (If the default config path doesn't exist yet, it's expected behavior.
+  // It will be created the first time now.)
+  if (!common::exists(_path))
+  {
+    configExists = false;
+    doc.Parse(this->dataPtr->config.c_str());
+    // Process each dialog
+    for (auto dialogElem = doc.FirstChildElement("dialog");
+      dialogElem != nullptr;
+      dialogElem = dialogElem->NextSiblingElement("dialog"))
+    {
+      if(dialogElem->Attribute("name") == this->objectName().toStdString())
+      {
+        value = dialogElem->Attribute(_attribute.c_str());
```

Comment:
Currently, if there are multiple dialog elements with the same name, the function will return the last of them. I think that behaviour is ok. Can you document it?